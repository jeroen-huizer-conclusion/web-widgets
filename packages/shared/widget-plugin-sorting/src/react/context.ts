import { Context, createContext, useContext } from "react";
import { SortStoreHost } from "../stores/SortStoreHost";
import { SortInstruction } from "../types/store";

export interface SortAPI {
    version: 1;
    host: SortStoreHost;
    initSortOrder?: SortInstruction[];
}

const SORT_PATH = "com.mendix.widgets.web.sortable.sortContext";

export function getGlobalSortContext(): Context<SortAPI | null> {
    return ((window as any)[SORT_PATH] ??= createContext<SortAPI | null>(null));
}

export function useSortAPI(): SortAPI {
    const api = useContext(getGlobalSortContext());
    if (api === null) {
        throw new Error("SortAPI is not available. Widget is out of parent context.");
    }
    return api;
}
