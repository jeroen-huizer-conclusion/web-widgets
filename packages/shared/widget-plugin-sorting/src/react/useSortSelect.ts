import { useSetup } from "@mendix/widget-plugin-mobx-kit/react/useSetup";
import { SingleSortController } from "../controllers/SingleSortController";
import { BasicSortStore, SortDirection } from "../types/store";

interface HookProps {
    emptyOptionCaption?: string;
    store: BasicSortStore;
}

interface Option {
    caption: string;
    value: string;
}

interface ControlProps {
    value: string | null;
    options: Option[];
    direction: SortDirection;
    onSelect: (value: string) => void;
    onDirectionClick: () => void;
}

export function useSortSelect(props: HookProps, store: BasicSortStore): ControlProps {
    const ctrl = useSetup(() => new SingleSortController({ store }));

    return {
        value: ctrl.selected,
        options: ctrl.options,
        direction: ctrl.direction,
        onSelect: ctrl.select,
        onDirectionClick: ctrl.toggleDirection
    };
}
