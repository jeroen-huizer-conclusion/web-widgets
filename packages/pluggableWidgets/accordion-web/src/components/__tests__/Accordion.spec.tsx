import "@testing-library/jest-dom";
import { createElement } from "react";
import { render, RenderResult } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Accordion, AccordionProps } from "../Accordion";

global.ResizeObserver = jest.fn().mockImplementation(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn()
}));

describe("Accordion", () => {
    const defaultProps: AccordionProps = {
        id: "id",
        class: "",
        style: { height: "500px" },
        tabIndex: 1,
        groups: [
            {
                header: "header",
                content: <span>content</span>,
                initiallyCollapsed: true,
                visible: true,
                loadContent: "always"
            },
            {
                header: "header2",
                content: <span>content2</span>,
                initiallyCollapsed: true,
                visible: false,
                loadContent: "always"
            }
        ],
        collapsible: true,
        singleExpandedGroup: true,
        generateHeaderIcon: jest.fn(),
        showGroupHeaderIcon: "right"
    };

    function renderAccordion(props: Partial<AccordionProps> = {}): RenderResult {
        return render(<Accordion {...defaultProps} {...props} />);
    }

    describe("collapsible AccordionGroupWrapper", () => {
        const user = userEvent.setup();

        it("gives the next accordion group button focus on arrow down key down", async () => {
            const groups = [...defaultProps.groups];
            groups[1].visible = true;

            const accordion = renderAccordion({ groups });
            const buttons = accordion.getAllByRole("button");
            buttons[0].focus();

            await user.keyboard("{ArrowDown}");

            expect(buttons[1]).toHaveFocus;
        });

        it("gives the previous accordion group button focus on arrow up key down", async () => {
            const groups = [...defaultProps.groups];
            groups[1].visible = true;

            const accordion = renderAccordion({ groups });

            const buttons = accordion.getAllByRole("button");
            buttons[buttons.length - 1].focus();

            await user.keyboard("{ArrowUp}");

            expect(buttons[0]).toHaveFocus();
        });

        it("gives the first accordion group button focus on Home key down", async () => {
            const groups = [
                ...defaultProps.groups,
                { header: "header3", content: <span>content3</span>, visible: true, loadContent: "always" as const }
            ];
            groups[1].visible = true;

            const accordion = renderAccordion({ groups });

            const buttons = accordion.getAllByRole("button");
            buttons[buttons.length - 1].focus();

            await user.keyboard("{Home}");

            expect(buttons[0]).toHaveFocus();
        });

        it("gives the last accordion group button focus on End key down", async () => {
            const groups = [
                ...defaultProps.groups,
                { header: "header3", content: <span>content3</span>, visible: true, loadContent: "always" as const }
            ];
            groups[1].visible = true;

            const accordion = renderAccordion({ groups });
            const buttons = accordion.getAllByRole("button");
            buttons[buttons.length - 1].focus();

            await user.keyboard("{End}");

            expect(buttons[buttons.length - 1]).toHaveFocus();
        });

        it("remains focus on the last accordion group button on arrow down key down", async () => {
            const groups = [
                ...defaultProps.groups,
                { header: "header3", content: <span>content3</span>, visible: true, loadContent: "always" as const }
            ];
            groups[1].visible = true;

            const accordion = renderAccordion({ groups });
            const buttons = accordion.getAllByRole("button");
            buttons[buttons.length - 1].focus();

            await user.keyboard("{ArrowDown}");

            expect(buttons[buttons.length - 1]).toHaveFocus();
        });
    });

    it("renders correctly without tabindex", () => {
        const accordion = renderAccordion({ tabIndex: undefined });

        expect(accordion.container).toMatchSnapshot();
    });

    describe("in collapsible & single expanded group mode", () => {
        const user = userEvent.setup();

        it("renders correctly", () => {
            const accordion = renderAccordion();

            expect(accordion.container).toMatchSnapshot();
        });

        it("expands a group", async () => {
            const accordion = renderAccordion();
            const button = accordion.getAllByRole("button")[1];

            await user.click(button);

            expect(accordion.container).toMatchSnapshot();
        });

        it("allows one group to be expanded only", async () => {
            const groups = [...defaultProps.groups];
            groups[1].visible = true;

            const accordion = renderAccordion({ groups });
            const buttons = accordion.getAllByRole("button");

            await user.click(buttons[0]);
            await user.click(buttons[buttons.length - 1]);

            expect(buttons[1]).toHaveFocus();
            expect(accordion.container).toMatchSnapshot();
        });

        it("collapses a group", async () => {
            const accordion = renderAccordion();
            const button = accordion.getAllByRole("button")[0];

            await user.click(button);
            await user.click(button);

            expect(accordion.container).toMatchSnapshot();
        });

        it("inits with group initially collapsed settings", () => {
            const groups = [...defaultProps.groups];
            groups[0].initiallyCollapsed = false;
            const accordion = renderAccordion({ groups });

            expect(accordion.container).toMatchSnapshot();
        });

        it("inits with not more than one group expanded", () => {
            const groups = [...defaultProps.groups].map(group => ({ ...group, initiallyCollapsed: false }));
            const accordion = renderAccordion({ groups });

            expect(accordion.container).toMatchSnapshot();
        });
    });

    describe("in collapsible & multiple expanded group mode", () => {
        const user = userEvent.setup();

        it("renders correctly", () => {
            const accordion = renderAccordion({ singleExpandedGroup: false });

            expect(accordion.container).toMatchSnapshot();
        });

        it("expands a group", async () => {
            const accordion = renderAccordion({ singleExpandedGroup: false });
            const button = accordion.getAllByRole("button")[0];

            await user.click(button);

            expect(accordion.container).toMatchSnapshot();
        });

        it("allows multiple groups to be expanded", async () => {
            const groups = [...defaultProps.groups];
            groups[1].visible = true;

            const accordion = renderAccordion({ singleExpandedGroup: false, groups });
            const buttons = accordion.getAllByRole("button");

            await user.click(buttons[0]);
            await user.click(buttons[buttons.length - 1]);

            expect(accordion.container).toMatchSnapshot();
        });

        it("collapses a group", async () => {
            const groups = [...defaultProps.groups];
            groups[1].visible = true;

            const accordion = renderAccordion({ singleExpandedGroup: false, groups });
            const buttons = accordion.getAllByRole("button");

            await user.click(buttons[0]);
            await user.click(buttons[buttons.length - 1]);
            await user.click(buttons[0]);

            expect(accordion.container).toMatchSnapshot();
        });

        it("inits with group initially collapsed settings", () => {
            const groups = [...defaultProps.groups].map(group => ({ ...group, initiallyCollapsed: false }));
            groups.push({
                header: "header3",
                content: <span>content3</span>,
                initiallyCollapsed: true,
                visible: true,
                loadContent: "always" as const
            });

            const accordion = renderAccordion({ singleExpandedGroup: false, groups });

            expect(accordion.container).toMatchSnapshot();
        });

        it("applies group collapsed value changes", () => {
            const accordion = renderAccordion({ singleExpandedGroup: false });
            const newGroups = [...defaultProps.groups];
            newGroups[1].collapsed = false;

            accordion.rerender(<Accordion {...defaultProps} singleExpandedGroup={false} groups={newGroups} />);

            expect(accordion.container).toMatchSnapshot();
        });
    });

    describe("not collapsible", () => {
        it("renders correctly", () => {
            const accordion = renderAccordion({ collapsible: false });

            expect(accordion.container).toMatchSnapshot();
        });
    });
});
