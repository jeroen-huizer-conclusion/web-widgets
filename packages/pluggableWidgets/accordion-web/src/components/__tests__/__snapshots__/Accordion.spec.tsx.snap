// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion in collapsible & multiple expanded group mode allows multiple groups to be expanded 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="true"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
    <section
      class="widget-accordion-group"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup1ContentWrapper"
          aria-disabled="false"
          aria-expanded="true"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup1HeaderButton"
          role="button"
          tabindex="0"
        >
          header2
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup1HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup1ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content2
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & multiple expanded group mode applies group collapsed value changes 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & multiple expanded group mode collapses a group 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
    <section
      class="widget-accordion-group"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup1ContentWrapper"
          aria-disabled="false"
          aria-expanded="true"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup1HeaderButton"
          role="button"
          tabindex="0"
        >
          header2
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup1HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup1ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content2
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & multiple expanded group mode expands a group 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="true"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & multiple expanded group mode inits with group initially collapsed settings 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="true"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup2ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup2HeaderButton"
          role="button"
          tabindex="0"
        >
          header3
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup2HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup2ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content3
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & multiple expanded group mode renders correctly 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & single expanded group mode allows one group to be expanded only 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
    <section
      class="widget-accordion-group"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup1ContentWrapper"
          aria-disabled="false"
          aria-expanded="true"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup1HeaderButton"
          role="button"
          tabindex="0"
        >
          header2
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup1HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup1ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content2
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & single expanded group mode collapses a group 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & single expanded group mode expands a group 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & single expanded group mode inits with group initially collapsed settings 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="true"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & single expanded group mode inits with not more than one group expanded 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion in collapsible & single expanded group mode renders correctly 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion not collapsible renders correctly 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="1"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="true"
          aria-expanded="false"
          class="widget-accordion-group-header-button"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Accordion renders correctly without tabindex 1`] = `
<div>
  <div
    class="widget-accordion"
    data-focusindex="0"
    id="id"
    style="height: 500px;"
  >
    <section
      class="widget-accordion-group widget-accordion-group-collapsed"
    >
      <header
        class="widget-accordion-group-header"
      >
        <div
          aria-controls="idAccordionGroup0ContentWrapper"
          aria-disabled="false"
          aria-expanded="false"
          class="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
          data-focusindex="0"
          id="idAccordionGroup0HeaderButton"
          role="button"
          tabindex="0"
        >
          header
        </div>
      </header>
      <div
        aria-labelledby="idAccordionGroup0HeaderButton"
        class="widget-accordion-group-content-wrapper"
        data-focusindex="0"
        id="idAccordionGroup0ContentWrapper"
        role="region"
      >
        <div
          class="widget-accordion-group-content"
        >
          <span>
            content
          </span>
        </div>
      </div>
    </section>
  </div>
</div>
`;
